### DISCUSSION TITLE
how to get localstorage value in laravel controller?

### DISCUSSION THREAD

**Discussion by webalpha107** [community/community#18404] [Other Feature Feedback, Questions, & Ideas] [answered]
how to get localstorage value in laravel controller?

**Comment by devcrafta** [comment, marked as answer]
I will suggest you to get the value from the localStorage with javascript using `localStorage.getItem('myItem')` and then store it as a cookie using `document.cookie = "name=value"`. You will now be able to access it from your laravel controller using `$_COOKIE['cookieName']`.

Example:

Javascript

//define a function to set cookies
```
function setCookie(name,value,days) {
   var expires = "";
   if (days) {
       var date = new Date();
       date.setTime(date.getTime() + (days*24*60*60*1000));
       expires = "; expires=" + date.toUTCString();
   }
   document.cookie = name + "=" + (value || "")  + expires + "; path=/";
}
```

//get your item from the localStorage
```
var myItem = localStorage.getItem('myItem');
setCookie('cookieName', myItem, 7);
```
PHP

```
$myItem = '';

if(isset($_COOKIE['cookieName'])) {
    $myItem = $_COOKIE['cookieName'];
}
```

**Comment by dev-718** [comment]
Hello, Astro.
Thank for your description.
Best regards.
